core/src/main/java/com/linecorp/armeria/common/util/TimeWindowPercentileSampler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.linecorp.armeria.common.util;
+
+import java.time.Duration;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.TimeWindowPercentileHistogram;
+
+/**
+ * Sample if the value is less than the percentile of the values in the last window.
+ */
+final class TimeWindowPercentileSampler implements Sampler<Long> {
+
+    private final float percentile;
+    private final long windowLengthMillis;
+
+    private final TimeWindowPercentileHistogram histogram;
+    private static final long SNAPSHOT_UPDATE_MILLIS = 1000L;
+    private long lastSnapshotMillis = 0L;
+    private HistogramSnapshot histogramSnapshot;
+
+    TimeWindowPercentileSampler(float percentile, long windowLengthMillis) {
+        this.percentile = percentile;
+        this.windowLengthMillis = windowLengthMillis;
+
+        final DistributionStatisticConfig distributionStatisticConfig =
+                DistributionStatisticConfig.builder()
+                                           .percentiles(percentile)
+                                           .percentilePrecision(2)
+                                           .minimumExpectedValue(1.0)
+                                           .maximumExpectedValue(Double.POSITIVE_INFINITY)
+                                           .expiry(Duration.ofMillis(windowLengthMillis))
+                                           .bufferLength(3)
```

Comment:
This also should be configurable.

Reply:
I believe the following helps with configuration

```
.merge(MoreMeters.distributionStatisticConfig());
```

So basically we are using the same buffer length we use for distribution statistics, which makes sense on my side. Any thoughts?